addons/point_of_sale/static/src/js/pos.web_editor.js

```diff
@@ -48,4 +49,18 @@ odoo.define('point_of_sale.editor', function (require) {
         }
     });
 
-});
\ No newline at end of file
+    FieldTextHtml.include({
+        // avoid '414 Request-URI Too Large' errors to
+        // /point_of_sale/field/customer_facing_display_template by
+        // filtering out biggest fields
+        get_datarecord: function () {
+            var datarecord = this._super();
+            if (this.model === 'pos.config') {
+                datarecord = _.omit(datarecord, function (val, key) {
+                    return _.isObject(val) || key === 'customer_facing_display_html';
+                });
+            }
+            return datarecord;
+        },
+    });
+});
```

Comment:
I'd put this in another file, which i'd put in the assets_backend bundle, so that the pos specific code of this file remains in the pos bundle.